Build issues

Hey :wave: I'm looking to use your crate to model an issue tracker and I'm pretty sure rose trees are exactly what I need :grin: 

Unfortunately, when I try to build the project I'm getting an error:
```
[nix-shell:~/Developer/rose_tree-rs]$ cargo build
    Updating crates.io index
   Compiling fixedbitset v0.1.9
   Compiling petgraph v0.1.18
error[E0034]: multiple applicable items in scope
   --> /home/haptop/.cargo/registry/src/github.com-1ecc6299db9ec823/petgraph-0.1.18/src/graph.rs:404:17
    |
404 |         assert!(Ix::max().index() == !0 || NodeIndex::end() != node_idx);
    |                 ^^^^^^^ multiple `max` found
    |
note: candidate #1 is defined in the trait `graph::IndexType`
   --> /home/haptop/.cargo/registry/src/github.com-1ecc6299db9ec823/petgraph-0.1.18/src/graph.rs:28:5
    |
28  |     fn max() -> Self;
    |     ^^^^^^^^^^^^^^^^^
    = help: to disambiguate the method call, write `graph::IndexType::max(...)` instead
note: candidate #2 is defined in the trait `std::cmp::Ord`
    = help: to disambiguate the method call, write `std::cmp::Ord::max(...)` instead

error[E0034]: multiple applicable items in scope
   --> /home/haptop/.cargo/registry/src/github.com-1ecc6299db9ec823/petgraph-0.1.18/src/graph.rs:438:17
    |
438 |         assert!(Ix::max().index() == !0 || EdgeIndex::end() != edge_idx);
    |                 ^^^^^^^ multiple `max` found
    |
note: candidate #1 is defined in the trait `graph::IndexType`
   --> /home/haptop/.cargo/registry/src/github.com-1ecc6299db9ec823/petgraph-0.1.18/src/graph.rs:28:5
    |
28  |     fn max() -> Self;
    |     ^^^^^^^^^^^^^^^^^
    = help: to disambiguate the method call, write `graph::IndexType::max(...)` instead
note: candidate #2 is defined in the trait `std::cmp::Ord`
    = help: to disambiguate the method call, write `std::cmp::Ord::max(...)` instead

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0034`.
error: could not compile `petgraph`.

To learn more, run the command again with --verbose.
```

My `rustc` version:
```
[nix-shell:~/Developer/rose_tree-rs]$ rustc --version
rustc 1.40.0 (73528e339 2019-12-16)
```

I'm going to keep digging, but wanted to post this as a first heads up :ok_hand:

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -11,4 +11,4 @@ homepage = "https://github.com/mitchmindtree/rose_tree-rs"
 
 
 [dependencies]
-petgraph = "0.1.11"
+petgraph = "0.5"

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -13,8 +13,10 @@
 pub extern crate petgraph;
 use petgraph as pg;
 
+use petgraph::graph::IndexType;
 pub use petgraph::graph::NodeIndex;
-use petgraph::graph::{DefIndex, IndexType};
+
+type DefIndex = u32;
 
 /// The PetGraph to be used internally within the RoseTree for storing/managing nodes and edges.
 pub type PetGraph<N, Ix> = pg::Graph<N, (), pg::Directed, Ix>;
@@ -72,7 +74,7 @@ pub struct Siblings<'a, Ix: IndexType> {
 
 /// A "walker" object that can be used to step through the children of some parent node.
 pub struct WalkChildren<Ix: IndexType> {
-    walk_edges: pg::graph::WalkEdges<Ix>,
+    walk_edges: pg::graph::WalkNeighbors<Ix>,
 }
 
 /// A "walker" object that can be used to step through the siblings of some child node.
@@ -180,8 +182,8 @@ where
         let parent = self.parent(node).expect("No parent node found");
 
         // For each of `node`'s children, set their parent to `node`'s parent.
-        let mut children = self.graph.walk_edges_directed(node, pg::Outgoing);
-        while let Some((child_edge, child_node)) = children.next_neighbor(&self.graph) {
+        let mut children = self.graph.neighbors_directed(node, pg::Outgoing).detach();
+        while let Some((child_edge, child_node)) = children.next(&self.graph) {
             self.graph.remove_edge(child_edge);
             self.graph.add_edge(parent, child_node, ());
         }
@@ -224,7 +226,7 @@ where
     ///
     /// Unlike the `Children` type, `WalkChildren` does not borrow the `RoseTree`.
     pub fn walk_children(&self, parent: NodeIndex<Ix>) -> WalkChildren<Ix> {
-        let walk_edges = self.graph.walk_edges_directed(parent, pg::Outgoing);
+        let walk_edges = self.graph.neighbors_directed(parent, pg::Outgoing).detach();
         WalkChildren { walk_edges }
     }
 
@@ -315,7 +317,7 @@ where
 {
     /// Fetch the next child index in the walk for the given `RoseTree`.
     pub fn next<N>(&mut self, tree: &RoseTree<N, Ix>) -> Option<NodeIndex<Ix>> {
-        self.walk_edges.next_neighbor(&tree.graph).map(|(_, n)| n)
+        self.walk_edges.next(&tree.graph).map(|(_, n)| n)
     }
 }
 

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -1,4 +1,4 @@
-//! 
+//!
 //! **rose_tree** is a rose tree (aka multi-way tree) data structure library.
 //!
 //! The most prominent type is [**RoseTree**](./struct.RoseTree.html) - a wrapper around [petgraph]
@@ -16,10 +16,8 @@ use petgraph as pg;
 pub use petgraph::graph::NodeIndex;
 use petgraph::graph::{DefIndex, IndexType};
 
-
 /// The PetGraph to be used internally within the RoseTree for storing/managing nodes and edges.
 pub type PetGraph<N, Ix> = pg::Graph<N, (), pg::Directed, Ix>;
-
 
 /// An indexable tree data structure with a variable and unbounded number of branches per node.
 ///
@@ -56,7 +54,6 @@ pub struct RoseTree<N, Ix: IndexType = DefIndex> {
     graph: PetGraph<N, Ix>,
 }
 
-
 /// An iterator that yeilds an index to the parent of the current child before the setting the
 /// parent as the new current child. This occurs recursively until the root index is yeilded.
 pub struct ParentRecursion<'a, N: 'a, Ix: IndexType> {
@@ -87,9 +84,10 @@ pub struct WalkSiblings<Ix: IndexType> {
 /// `RoseTree`'s API ensures that it always has a "root" node and that its index is always 0.
 pub const ROOT: usize = 0;
 
-
-impl<N, Ix> RoseTree<N, Ix> where Ix: IndexType {
-
+impl<N, Ix> RoseTree<N, Ix>
+where
+    Ix: IndexType,
+{
     /// Create a new `RoseTree` along with some root node.
     /// Returns both the `RoseTree` and an index into the root node in a tuple.
     pub fn new(root: N) -> (Self, NodeIndex<Ix>) {
@@ -101,7 +99,7 @@ impl<N, Ix> RoseTree<N, Ix> where Ix: IndexType {
     pub fn with_capacity(nodes: usize, root: N) -> (Self, NodeIndex<Ix>) {
         let mut graph = PetGraph::with_capacity(nodes, nodes);
         let root = graph.add_node(root);
-        (RoseTree { graph: graph }, root)
+        (RoseTree { graph }, root)
     }
 
     /// The total number of nodes in the RoseTree.
@@ -173,7 +171,6 @@ impl<N, Ix> RoseTree<N, Ix> where Ix: IndexType {
     ///
     /// Note: this method may shift other node indices, invalidating previously returned indices!
     pub fn remove_node(&mut self, node: NodeIndex<Ix>) -> Option<N> {
-
         // Check if an attempt to remove the root node has been made.
         if node.index() == ROOT || self.graph.node_weight(node).is_none() {
             return None;
@@ -210,7 +207,10 @@ impl<N, Ix> RoseTree<N, Ix> where Ix: IndexType {
     ///
     /// The returned iterator yields `NodeIndex<Ix>`s.
     pub fn parent_recursion(&self, child: NodeIndex<Ix>) -> ParentRecursion<N, Ix> {
-        ParentRecursion { rose_tree: self, child: child }
+        ParentRecursion {
+            rose_tree: self,
+            child,
+        }
     }
 
     /// An iterator over all nodes that are children to the node at the given index.
@@ -225,7 +225,7 @@ impl<N, Ix> RoseTree<N, Ix> where Ix: IndexType {
     /// Unlike the `Children` type, `WalkChildren` does not borrow the `RoseTree`.
     pub fn walk_children(&self, parent: NodeIndex<Ix>) -> WalkChildren<Ix> {
         let walk_edges = self.graph.walk_edges_directed(parent, pg::Outgoing);
-        WalkChildren { walk_edges: walk_edges }
+        WalkChildren { walk_edges }
     }
 
     /// An iterator over all nodes that are siblings to the node at the given index.
@@ -233,7 +233,10 @@ impl<N, Ix> RoseTree<N, Ix> where Ix: IndexType {
     /// The returned iterator yields `NodeIndex<Ix>`s.
     pub fn siblings(&self, child: NodeIndex<Ix>) -> Siblings<Ix> {
         let maybe_siblings = self.parent(child).map(|parent| self.children(parent));
-        Siblings { child: child, maybe_siblings: maybe_siblings }
+        Siblings {
+            child,
+            maybe_siblings,
+        }
     }
 
     /// A "walker" object that may be used to step through the siblings of the given child node.
@@ -241,65 +244,99 @@ impl<N, Ix> RoseTree<N, Ix> where Ix: IndexType {
     /// Unlike the `Siblings` type, `WalkSiblings` does not borrow the `RoseTree`.
     pub fn walk_siblings(&self, child: NodeIndex<Ix>) -> WalkSiblings<Ix> {
         let maybe_walk_children = self.parent(child).map(|parent| self.walk_children(parent));
-        WalkSiblings { child: child, maybe_walk_children: maybe_walk_children }
+        WalkSiblings {
+            child,
+            maybe_walk_children,
+        }
     }
-
 }
 
-
-impl<N, Ix> ::std::ops::Index<NodeIndex<Ix>> for RoseTree<N, Ix> where Ix: IndexType {
+impl<N, Ix> ::std::ops::Index<NodeIndex<Ix>> for RoseTree<N, Ix>
+where
+    Ix: IndexType,
+{
     type Output = N;
     fn index(&self, index: NodeIndex<Ix>) -> &N {
         &self.graph[index]
     }
 }
 
-impl<N, Ix> ::std::ops::IndexMut<NodeIndex<Ix>> for RoseTree<N, Ix> where Ix: IndexType {
+impl<N, Ix> ::std::ops::IndexMut<NodeIndex<Ix>> for RoseTree<N, Ix>
+where
+    Ix: IndexType,
+{
     fn index_mut(&mut self, index: NodeIndex<Ix>) -> &mut N {
         &mut self.graph[index]
     }
 }
 
-
-impl<'a, Ix> Iterator for Siblings<'a, Ix> where Ix: IndexType {
+impl<'a, Ix> Iterator for Siblings<'a, Ix>
+where
+    Ix: IndexType,
+{
     type Item = NodeIndex<Ix>;
     fn next(&mut self) -> Option<NodeIndex<Ix>> {
-        let Siblings { child, ref mut maybe_siblings } = *self;
+        let Siblings {
+            child,
+            ref mut maybe_siblings,
+        } = *self;
         maybe_siblings.as_mut().and_then(|siblings| {
             siblings.next().and_then(|sibling| {
-                if sibling != child { Some(sibling) } else { siblings.next() }
+                if sibling != child {
+                    Some(sibling)
+                } else {
+                    siblings.next()
+                }
             })
         })
     }
 }
 
-impl<'a, N, Ix> Iterator for ParentRecursion<'a, N, Ix> where Ix: IndexType {
+impl<'a, N, Ix> Iterator for ParentRecursion<'a, N, Ix>
+where
+    Ix: IndexType,
+{
     type Item = NodeIndex<Ix>;
     fn next(&mut self) -> Option<NodeIndex<Ix>> {
-        let ParentRecursion { ref mut child, ref rose_tree } = *self;
-        rose_tree.parent(*child).map(|parent| { *child = parent; parent })
+        let ParentRecursion {
+            ref mut child,
+            ref rose_tree,
+        } = *self;
+        rose_tree.parent(*child).map(|parent| {
+            *child = parent;
+            parent
+        })
     }
 }
 
-
-impl<Ix> WalkChildren<Ix> where Ix: IndexType {
+impl<Ix> WalkChildren<Ix>
+where
+    Ix: IndexType,
+{
     /// Fetch the next child index in the walk for the given `RoseTree`.
     pub fn next<N>(&mut self, tree: &RoseTree<N, Ix>) -> Option<NodeIndex<Ix>> {
         self.walk_edges.next_neighbor(&tree.graph).map(|(_, n)| n)
     }
 }
 
-impl<Ix> WalkSiblings<Ix> where Ix: IndexType {
+impl<Ix> WalkSiblings<Ix>
+where
+    Ix: IndexType,
+{
     /// Fetch the next sibling index in the walk for the given `RoseTree`.
     pub fn next<N>(&mut self, tree: &RoseTree<N, Ix>) -> Option<NodeIndex<Ix>> {
-        let WalkSiblings { child, ref mut maybe_walk_children } = *self;
+        let WalkSiblings {
+            child,
+            ref mut maybe_walk_children,
+        } = *self;
         maybe_walk_children.as_mut().and_then(|walk_children| {
-            walk_children.next(tree).and_then(|sibling| if child != sibling {
-                Some(sibling)
-            } else {
-                walk_children.next(tree)
+            walk_children.next(tree).and_then(|sibling| {
+                if child != sibling {
+                    Some(sibling)
+                } else {
+                    walk_children.next(tree)
+                }
             })
         })
     }
 }
-

